docs/rules/idiomatic/use-some-for-output-vars.md

```diff
@@ -0,0 +1,102 @@
+# use-some-for-output-vars
+
+**Summary**: Use `some` to declare output variables
+
+**Category**: Idiomatic
+
+**Avoid**
+```rego
+package policy
+
+allow {
+    userinfo := data.users[id]
+    # ...
+}
+```
+
+**Prefer**
+```rego
+allow {
+    some id
+    userinfo := data.users[id]
+    # ...
+}
+
+# alternatively, and arguably more idiomatic:
+allow {
+    some id, userinfo in data.users
+    # ...
+}
+```
+
+## Rationale
+
+An interesting, and likely unfamiliar aspect of Rego for developers coming from other languages, is the concept of
+[unification](https://en.wikipedia.org/wiki/Unification_(computer_science)). Unification happens not just explicitly via
+the unification operator (`=`), but is an integral part of Rego. In the context of this rule, unification means that a
+variable can either be an _input_ or an _output_. What does that mean?
```

Comment:
I suppose some of that would make sense to be more explicitly and thoroughly explained in the OPA docs, too.

Reply:
Agreed.